change/@ni-eslint-config-angular-718354fd-7ab1-4395-ac53-e8505b9622bf.json

```diff
@@ -0,0 +1,7 @@
+{
+  "type": "major",
+  "comment": "Update to Angular 18",
+  "packageName": "@ni/eslint-config-angular",
```

Comment:
FYI, #157 exists to track this work. It links to a few related issues: could you evaluate fixing at the same time?

Reply:
One of the linked issues is migrating to eslint 9 which I think we should do in a follow-up instead after the Angular 18 upgrade. I suspect and commented on https://github.com/ni/javascript-styleguide/issues/155https://github.com/ni/javascript-styleguide/issues/155 that eslint 9 migration probably will take some effort.

Reply:
Treating the ESLint 9 update as a separate task, but I'll link/fix the other issues with this PR.